.azure-pipelines/main-pipeline.yml

```diff
@@ -1,21 +1,20 @@
-# Disable the pipeline for individual commits except on `main` merges.
-# The pipeline requires a Pull Request to run.
+# Disable the pipeline for individual commits except on `main` merges, and on version tags.
 trigger:
-  - main
-
-parameters:
-  - name: performRelease
-    displayName: Release Production Extension
-    type: boolean
-    default: false
-  - name: releaseSemver
-    displayName: Release Production semver
-    type: string
-    default: Patch
-    values:
-      - Patch
-      - Minor
-      - Major
+  branches:
+    include:
+      - main
+      - refs/tags/v* # Only version tags
+  tags:
+    exclude:
+      - '*' # Exclude all other tags
+
+# The pipeline requires a PR to run on branches.
+pr:
+  branches:
+    include:
+      - '*'
+    exclude:
+      - refs/tags/* # Exclude all other tags
```

Comment:
Those are rules to require PRs to be created for a pipeline to run on a branch. They look like this in the Azure DevOps UI:

<img width="428" alt="image" src="https://github.com/DataDog/datadog-ci-azure-devops/assets/9317502/ae23185e-f95c-408e-ada1-32e7bc091837">

This rule means that:
- Any branches can be accepted.
- But tags that are pointing to feature branch commits are excluded

Reply:
I'm not sure what the difference is between the `trigger` and `pr` sections are used for: It seems like they are referring to exact opposite things, and it's unclear what pipelines they kick off. Do they trigger the same thing? If they trigger different things, what do they each trigger?

Reply:
Yes they trigger the same thing: the `main-pipeline.yml` file. The same pipeline simply runs on different events.

And the main pipeline is composed of:
- The `Build` stage
- The `ReleaseDev` stage (which is forbidden to run on tags because it already runs on commits)
- The `ReleasePublic` stage (which only runs on "official" tags, aka. version tags)

Reply:
Gotcha - not to be too picky, but is there a way the pipeline could be written in a way that's like:

[when: tag]
- Build
- ReleasePublic

[when: pr that's not a tag]
- Build
- ReleaseDev

The current structure looks a bit more like:

[when: (main branch or tag) or (pr that's not a tag)]
- Build
- [if not tag] ReleaseDev
- [if tag] ReleasePublic

I feel like it might be slightly more intuitive with the first structure, and it would involve fewer conditionals throughout the file. Feel free to ignore this if the structuring of the file isn't possible.

Reply:
I feel you but it's not possible to have such structure... Otherwise, we'd have to "always run the pipeline" and add more conditionals for each stage, but you can't make the difference between a branch and a PR (outside of using the top-level `pr` section).

The other solution would be to split the `main-pipeline.yml` into two files.

Reply:
ah, okay, i was hoping, but if it's not possible that's fine

Reply:
When I explained this, I realized that the E2E pipeline was **always** being run when the `main-pipeline.yml` finishes, which wasn't ideal. I tried to configure the E2E pipeline to only start if the `ReleaseDev` stage had been run in the main pipeline, but it's not possible.

I could have solved this by also adding a `${{ if not(startsWith(variables['Build.SourceBranch'], 'refs/tags/')) }}` condition in the E2E pipeline too, but I felt like it was becoming messy.

So in the end, I went ahead and split the `main-pipeline.yml` into:
- `dev-pipeline.yml`, which only contains `Build` and `ReleaseDev`
- `release-pipeline.yml`, which only contains `Build` and `ReleasePublic`

So now, both files only contain the relevant top-level conditions.